### PR TITLE
fix(Core/Pets): Mob "soloed" by pet should not grant exp/loot to its …

### LINKED ISSUE #11528: [Hunter] Pet solo killing grants EXP, loots etc to Master
https://github.com/chromiecraft/chromiecraft/issues/3408

### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

Attacking with a pet only whilst the hunter just sits and does nothing, still grants XP, loot, and the ability to skin mobs to the hunter.

### Expected Blizzlike Behaviour

Normally if a pet is the only one doing damage, no XP should be rewarded and no drops provided. 

### Source

As per chat in Chromiecraft Discord with Nefertum, Nyeriah, etc in GM channels.

### Steps to reproduce the problem

1. Summon hunter pet
2. Put pet in aggressive mode
3. Sit back, relax and watch the XP flow

### Extra Notes

Just to add you can also earn quest kill progression (for example, kill 10 whelps) by using only the pet to kill as well. Believe this shouldn't be the case?

Edit:
The PR is already out but I think it's interesting how you can find info while looking for something else entirely.

>Just started the quest involving this guy, checked around, he wasn't there. went afk, got some food, came back and my pet had killed him, and since I hadn't done and dmg to him I couldn't loot him.
Bad pet
/punish ([source](https://tbc.wowhead.com/npc=10816/wandering-skeleton#comments:id=293131))

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/343732ee0beef702181009e26c41722b87f9cbe2

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
…master.

Fixes #11528

## Issues Addressed:

- Closes #11528
- Closes https://github.com/chromiecraft/chromiecraft/issues/3408

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Summon hunter pet
Put pet in aggressive mode
Sit back, relax and watch the XP

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
It works but it's making the player need to deal a percentage of damage before they're eligible for loot/XP (40-50%, maybe?) which I think it's bad for Beast Mastery hunters that can have their pets normally deal at least 50% or more of their total damage, varying between players of course.

Using Mend Pet or pet abilities do not make the player eligible. But I don't know what's blizzlike in that case.

**Comment by UltraNix** [comment]
You need to do at least 50% + 1 damage to get loot/exp - it's blizzlike.

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by temperrr** [comment]
Level 60 BM hunters pet kill monsters so quickly that this change would make them have no loot at all. This does not seem right.

**Comment by Nyeriah** [comment]
Yes, I wanted to try this on retail but I pretty much remember you don't need to do much in order to get credit, definitely not 50%. Most of the time BM hunters won't get any loot or experience yes

**Comment by T1ti** [comment]
I tested it and 1 tick of serpent sting was enough to give me loot

**Comment by Nyeriah** [comment]
> I tested it on classic and 1 tick of serpent sting was enough to give me loot

Thanks for confirming @T1ti , I discussed this with @temperrr and that was our impression as well.

**Comment by UltraNix** [comment]
I said that many times - classic is not vanilla (and especially is not wotlk).
https://github.com/TrinityCore/TrinityCore/issues/18758
https://github.com/TrinityCore/TrinityCore/issues/12478
https://github.com/TrinityCore/TrinityCore/issues/8549
https://web.archive.org/web/20160917231252/https://eu.battle.net/forums/en/wow/topic/8819991938
https://www.mmo-champion.com/threads/1798474-Sooo-Having-a-pet-makes-you-lose-loot

**Comment by UltraNix** [comment]
@Gultask @temperrr Please, test it again.

**Comment by Gultask** [comment]
Tested. The player now needs to land the killing blow. `.damage` until 1hp remains, send pet, don't get credit.

**Comment by UltraNix** [comment]
@Gultask Should be okay now.

**Comment by Gultask** [comment]
Looks great! a single point of damage is enough to give credit.
Tested loot, XP, reputation and quest credit.

**Comment by temperrr** [comment]
Thanks for the test Gultask <3

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -215,8 +215,8 @@ bool TemporaryThreatModifierEvent::Execute(uint64 /*e_time*/, uint32 /*p_time*/)
     return true;
 }
 
-Creature::Creature(bool isWorldObject): Unit(isWorldObject), MovableMapObject(), m_groupLootTimer(0), lootingGroupLowGUID(0), m_PlayerDamageReq(0), m_lootRecipientGroup(0),
-    m_corpseRemoveTime(0), m_respawnTime(0), m_respawnDelay(300), m_corpseDelay(60), m_wanderDistance(0.0f), m_boundaryCheckTime(2500),
+Creature::Creature(bool isWorldObject): Unit(isWorldObject), MovableMapObject(), m_groupLootTimer(0), lootingGroupLowGUID(0), _playerDamageReq(0), _damagedByPlayer(false),
+    m_lootRecipientGroup(0), m_corpseRemoveTime(0), m_respawnTime(0), m_respawnDelay(300), m_corpseDelay(60), m_wanderDistance(0.0f), m_boundaryCheckTime(2500),
```
Comment (acidmanifesto) [inline src/server/game/Entities/Creature/Creature.cpp:219]: This is tossing a error on the compile testing. I am guess order?
image

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by UltraNix** [comment]
@acidmanifesto Should be okay.

File: src/server/game/Entities/Creature/Creature.cpp [hunk 2]
```diff
@@ -215,8 +215,8 @@ bool TemporaryThreatModifierEvent::Execute(uint64 /*e_time*/, uint32 /*p_time*/)
     return true;
 }
 
-Creature::Creature(bool isWorldObject): Unit(isWorldObject), MovableMapObject(), m_groupLootTimer(0), lootingGroupLowGUID(0), m_PlayerDamageReq(0), m_lootRecipientGroup(0),
-    m_corpseRemoveTime(0), m_respawnTime(0), m_respawnDelay(300), m_corpseDelay(60), m_wanderDistance(0.0f), m_boundaryCheckTime(2500),
+Creature::Creature(bool isWorldObject): Unit(isWorldObject), MovableMapObject(), m_groupLootTimer(0), lootingGroupLowGUID(0), _playerDamageReq(0),
+    m_lootRecipientGroup(0), _damagedByPlayer(false), m_corpseRemoveTime(0), m_respawnTime(0), m_respawnDelay(300), m_corpseDelay(60), m_wanderDistance(0.0f), m_boundaryCheckTime(2500),
```
Comment (acidmanifesto) [inline src/server/game/Entities/Creature/Creature.cpp:219]: lets try moving _playerDamageReq(0) to after  m_lootRecipientGroup(0) and see how it plays from there.
image

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by UltraNix** [comment]
Try now.

**Review by acidmanifesto — APPROVED** [review]

**Comment by acidmanifesto** [comment]
so far, the compile testing is crying that it needs to be moved over more to the right. at this point this is literally going to be commit and hope it goes green on the test. 

image

**Comment by acidmanifesto** [comment]
Thank you for the PR

**Pull request merged**